--- Makefile.am 	2016-04-08 13:55:04.000000000 +0200
+++ Makefile.am	2016-04-08 14:10:25.000000000 +0200
@@ -43,6 +43,10 @@
     src/bus.h \
     src/tcpmux.h
 
+if OFI
+nninclude_HEADERS += src/ofi.h
+endif #OFI
+
 lib_LTLIBRARIES = libnanomsg.la
 
 NANOMSG_CORE = \
@@ -306,6 +310,22 @@
     src/transports/ws/sha1.h \
     src/transports/ws/sha1.c
 
+TRANSPORTS_OFI = \
+    src/transports/ofi/sofi.h \
+    src/transports/ofi/sofi.c \
+    src/transports/ofi/bofi.h \
+    src/transports/ofi/bofi.c \
+    src/transports/ofi/cofi.h \
+    src/transports/ofi/cofi.c \
+    src/transports/ofi/ofiw.h \
+    src/transports/ofi/ofiw.c \
+    src/transports/ofi/ofiapi.h \
+    src/transports/ofi/ofiapi.c \
+    src/transports/ofi/ofimr.h \
+    src/transports/ofi/ofimr.c \
+    src/transports/ofi/ofi.h \
+    src/transports/ofi/ofi.c 
+
 TRANSPORTS_TCPMUX = \
     src/transports/tcpmux/atcpmux.h \
     src/transports/tcpmux/atcpmux.c \
@@ -325,6 +345,10 @@
     $(TRANSPORTS_TCP) \
     $(TRANSPORTS_WS) \
     $(TRANSPORTS_TCPMUX)
+    
+if OFI
+NANOMSG_TRANSPORTS += $(TRANSPORTS_OFI)
+endif #OFI
 
 libnanomsg_la_SOURCES = \
     src/transport.h \
