--- src/core/global.c	2015-11-20 18:16:48.000000000 +0100
+++ src/core/global.c	2015-11-28 21:21:59.000000000 +0100
@@ -49,6 +49,7 @@
 #include "../transports/tcp/tcp.h"
 #include "../transports/ws/ws.h"
 #include "../transports/tcpmux/tcpmux.h"
+#include "../transports/ofi/ofi.h"
 
 #include "../protocols/pair/pair.h"
 #include "../protocols/pair/xpair.h"
@@ -257,6 +258,7 @@
     nn_global_add_transport (nn_tcp);
     nn_global_add_transport (nn_ws);
     nn_global_add_transport (nn_tcpmux);
+    nn_global_add_transport (nn_ofi);
 
     /*  Plug in individual socktypes. */
     nn_global_add_socktype (nn_pair_socktype);
