--- Makefile.am	2015-11-20 18:16:48.000000000 +0100
+++ Makefile.am	2015-12-14 20:03:56.000000000 +0100
@@ -35,6 +35,7 @@
     src/ipc.h \
     src/tcp.h \
     src/ws.h \
+    src/ofi.h \
     src/pair.h \
     src/pubsub.h \
     src/reqrep.h \
@@ -306,6 +311,22 @@
     src/transports/ws/sha1.h \
     src/transports/ws/sha1.c
 
+TRANSPORTS_OFI = \
+    src/transports/ofi/hlapi.h \
+    src/transports/ofi/hlapi.c \
+    src/transports/ofi/sofi_in.h \
+    src/transports/ofi/sofi_in.c \
+    src/transports/ofi/sofi_out.h \
+    src/transports/ofi/sofi_out.c \
+    src/transports/ofi/sofi.h \
+    src/transports/ofi/sofi.c \
+    src/transports/ofi/bofi.h \
+    src/transports/ofi/bofi.c \
+    src/transports/ofi/cofi.h \
+    src/transports/ofi/cofi.c \
+    src/transports/ofi/ofi.h \
+    src/transports/ofi/ofi.c 
+
 TRANSPORTS_TCPMUX = \
     src/transports/tcpmux/atcpmux.h \
     src/transports/tcpmux/atcpmux.c \
@@ -324,7 +341,8 @@
     $(TRANSPORTS_IPC) \
     $(TRANSPORTS_TCP) \
     $(TRANSPORTS_WS) \
-    $(TRANSPORTS_TCPMUX)
+    $(TRANSPORTS_TCPMUX) \
+    $(TRANSPORTS_OFI)
 
 libnanomsg_la_SOURCES = \
     src/transport.h \
@@ -339,7 +357,7 @@
 pkgconfigdir = $(libdir)/pkgconfig
 pkgconfig_DATA = libnanomsg.pc
 
-libnanomsg_la_LDFLAGS = -no-undefined -version-info @NN_LIBTOOL_VERSION@
+libnanomsg_la_LDFLAGS = -lfabric -no-undefined -version-info @NN_LIBTOOL_VERSION@
 libnanomsg_la_CFLAGS = -DNN_EXPORTS
 
 ################################################################################
